File: 953-reverse-only-letters/reverse-only-letters.java
class Solution {
    public String reverseOnlyLetters(String s) {
        char[] str = s.toCharArray();
        int start = 0;
        int end = str.length - 1;
        while (start < end){
if(Character.isLetter(str[start])){
    if(Character.isLetter(str[end])){
char temp = str[end];
str[end] = str[start];
str[start] = temp;
start++;
end--;
    }
    else{
        end--;
    }
}
else{
    start++;
}
        }    
        String res = new String(str);
return res;
    }
}

File: 953-reverse-only-letters/reverse-only-letters.py
class Solution:
    def reverseOnlyLetters(self, s: str) -> str:
        str = list(s)
        start = 0
        end = len(str)-1
        while start < end:
            if str[start].isalpha():
                if str[end].isalpha():
                    str[end],str[start] = str[start],str[end]
                    start += 1
                    end -= 1
                else:
                    end -= 1
            else:
                start += 1
        return ''.join(str)